public/app/features/alerting/unified/components/rule-editor/alert-rule-form/simplifiedRouting/route-settings/RouteSettings.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useState } from 'react';
+import { useFormContext } from 'react-hook-form';
+
+import { Field, FieldValidationMessage, InputControl, MultiSelect, Stack, Switch, Text, useStyles2 } from '@grafana/ui';
+import { useAlertmanagerConfig } from 'app/features/alerting/unified/hooks/useAlertmanagerConfig';
+import { useAlertmanager } from 'app/features/alerting/unified/state/AlertmanagerContext';
+import { RuleFormValues } from 'app/features/alerting/unified/types/rule-form';
+import {
+  commonGroupByOptions,
+  mapMultiSelectValueToStrings,
+  stringToSelectableValue,
+  stringsToSelectableValues,
+} from 'app/features/alerting/unified/utils/amroutes';
+
+import { getFormStyles } from '../../../../notification-policies/formStyles';
+import { TIMING_OPTIONS_DEFAULTS } from '../../../../notification-policies/timingOptions';
+
+import { RouteTimmings } from './RouteTimings';
+
+export interface RoutingSettingsProps {
+  alertManager: string;
+}
+export const RoutingSettings = ({ alertManager }: RoutingSettingsProps) => {
+  const formStyles = useStyles2(getFormStyles);
+  const {
+    control,
+    watch,
+    register,
+    formState: { errors },
+  } = useFormContext<RuleFormValues>();
+  const [groupByOptions, setGroupByOptions] = useState(stringsToSelectableValues([]));
+  const { groupBy, groupIntervalValue, groupWaitValue, repeatIntervalValue } = useGetDefaultsForRoutingSettings();
+  return (
+    <Stack direction="column">
+      <Stack direction="row" gap={1} alignItems="center" justifyContent="space-between">
+        <Field label="Override grouping">
+          <Switch id="override-grouping-toggle" {...register(`contactPoints.${alertManager}.overrideGrouping`)} />
+        </Field>
+        {!watch(`contactPoints.${alertManager}.overrideGrouping`) && (
```

Comment:
nit: Can we hoist all of the `watch` calls to variables here? I find that easier to keep track of what form changes re-render the component.